presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -546,6 +556,52 @@ private RelationPlan planLateralJoin(Join join, RelationPlan leftPlan, Lateral l
         return new RelationPlan(planBuilder.getRoot(), analysis.getScope(join), outputSymbols);
     }
 
+    private TranslationMap filterExpressionTranslationMap(RelationPlan leftPlan, RelationPlan rightPlan, PlanBuilder leftPlanBuilder, PlanBuilder rightPlanBuilder, Join join)
+    {
+        List<Symbol> rewriterOutputSymbols = ImmutableList.<Symbol>builder()
+                .addAll(leftPlan.getFieldMappings())
+                .addAll(rightPlan.getFieldMappings())
+                .build();
+
+        PlanNode root = new JoinNode(
```

Comment:
This doesn't seem right. Planning a join + lateral should not involve a join node. After all, we're modeling it as a specialized "apply + join" operation. Can you explain why you think this is needed?

Reply:
It looks like a egg-chicken-egg problem :). We do the same for join. We create join node to rewrite expression to create join. As I understand it is required to create a `TranslationMap`, because it is needed by `RelationPlan`. However, `PlanNode` from `RelationPlan` is not used in `TranslationMap`. I think we should refactor that and remove a need of `PlanNode` for `TranslationMap`.

Also I think this code should belong `SubqueriesPlanner`. `SubqueriesPlanner` should do proper expression rewrites.

Reply:
@kokosing you're right. It's just a dummy used to create a `TranslationMap` and not used in the resulting plan. It was inspired by `RelationPlanner.visitJoin()`

Reply:
>  I think we should refactor that and remove a need of PlanNode for TranslationMap.

Let's not make structural changes quite yet. I have work in progress to rewrite RelationPlanner and friends so we can split the AST from the IR entirely, and it touches all that you're talking about. @kasiafi's addition is fine, though -- I can adapt it easily since it's an incremental change.

Reply:
Yes, apparently `ValuesNode` works fine and even with empty `rows`. That looks much better as a dummy, thanks.